chain/network/src/types.rs

```diff
@@ -385,6 +386,112 @@ impl From<AnnounceAccount> for network_proto::AnnounceAccount {
     }
 }
 
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub enum DirectMessageBody {
+    BlockApproval(AccountId, CryptoHash, Signature),
+    ForwardTx(SignedTransaction),
+}
+
+impl DirectMessageBody {
+    pub fn hash(&self) -> CryptoHash {
+        match &self {
+            DirectMessageBody::BlockApproval(_, hash, _) => hash.clone(),
+            DirectMessageBody::ForwardTx(tx) => tx.get_hash(),
+        }
+    }
+}
+
+#[derive(Message)]
+pub struct RawDirectMessage {
+    pub account_id: AccountId,
+    pub body: DirectMessageBody,
+}
+
+impl RawDirectMessage {}
+
+/// DirectMessage represent a package that will travel the network towards a specific account id.
+/// It contains the peer_id and signature from the original sender. Every intermediate peer in the
+/// route must verify that this signature is valid otherwise previous sender of this package should
+/// be banned. If the final receiver of this package finds that the body is invalid the original
+/// sender of the package should be banned instead. (Notice that this peer might not be connected
+/// to us)
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub struct DirectMessage {
```

Comment:
a bit confusing name - I would expect DirectMessage to be message that doesn't get routed but goes directly.
`DirectedMessage` may be? or `RoutedMessage`

Reply:
Yes, in my mind it was quite clear what I was trying to say with **direct**, but you are right it's confusing. I will change it.